### PR TITLE
Include filterblock time in scan time metric for Coalescing readers

### LINKED ISSUE #6303: [BUG] Coalescing readers don't include filterblock time in scan time metric
**Describe the bug**
The coalescing readers ( at least the parquet one - need to check the others), don't include the time it takes to filter block in the scan time because it is called before the hasNext() function, which is where we track the scan time.

We need to make sure that time is tracked and accounted for.
Note for the multi-threaded reader the filter time happens in the background thread.  Should also check the per file reader to make sure its doing the right thing there, which I would guess it is.

### PR DESCRIPTION
**Pull request by thirtiseven**
The coalescing readers (parquet, ORC and Avro), don't include the time it takes to filter block in the scan time because it is called before the `hasNext()` function, which is where we track the scan time.

This PR includes filter block time in "scan time" metric for Coalescing readers. 

I checked that filter time is included in scan time in Spark, and also included in the plugin's per-file and multi-thread reader.

I add a temporary "scan time 2" metric locally to track time in filter block for tests. The metric will > 0ms only when the read type is Coalescing for parquet, ORC and Avro.

Note that there is no metric "scan time" in both spark and plugin when using DataSource V2, so it needs to check the existence of "scan time" metric.

Fixes #6303

Signed-off-by: thirtiseven <ntlihy@gmail.com>

### CODE REVIEW

**Review by jlowe — DISMISSED** [review]

**Comment by thirtiseven** [comment]
build

**Comment by thirtiseven** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala [hunk 1]
```diff
@@ -329,6 +330,9 @@ case class GpuOrcMultiFilePartitionReaderFactory(
             OrcSchemaWrapper(orcPartitionReaderContext.updatedReadSchema),
             OrcExtraInfo(orcPartitionReaderContext.requestedMapping)))
     }
+    if (metrics.contains("scanTime")) {
+      metrics("scanTime") += TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - currentTime)
+    }
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala:335]: This does an unnecessary double-lookup in the map, once for the contains and once again to do the update.
```suggestion
    metrics.get("scanTime").foreach {
      _ += TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - currentTime)
    }
```
Applies to the other places contains is used.

Comment (thirtiseven) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala]: Done, thanks!

**Comment by thirtiseven** [comment]
build

**Comment by pxLi** [comment]
build

**Review by jlowe — APPROVED** [review]
This PR is still marked Draft.  Intentional?

**Pull request merged**